openslides/agenda/projector.py

```diff
@@ -69,31 +69,63 @@ def get_collection_elements_required_for_this(self, collection_element, config_e
             output.extend(self.get_requirements_as_collection_elements(config_entry))
         return output
 
+    def update_data(self):
+        data = {}
+        if Item.objects.filter(pk=self.config_entry.get('id')).exists():
+            data['agenda_item_id'] = Item.objects.get(pk=self.config_entry.get('id')).item_number
+        return data
 
-class CurrentListOfSpeakersSlide(ProjectorElement):
-    """
-    Slide for the current list of speakers.
 
-    Nothing special to check.
+class CurrentListOfSpeakersMetaClass(ProjectorElement):
     """
-    name = 'agenda/current-list-of-speakers'
+    Main class for the list of speaker slides.
 
+    """
     def get_requirements(self, config_entry):
-        pk = config['projector_currentListOfSpeakers_reference']
-        if pk is not None:
-            # List of speakers slide.
-            try:
-                item = Item.objects.get(pk=pk)
-            except Item.DoesNotExist:
-                # Item does not exist. Just do nothing.
-                pass
-            else:
-                yield item
-                for speaker in item.speakers.filter(end_time=None):
-                    # Yield current speaker and next speakers
-                    yield speaker.user
+        items = self.get_agenda_items(config['projector_currentListOfSpeakers_reference'])
+        for item in items:
+            yield item
+            for speaker in item.speakers.filter(end_time=None):
+                yield speaker.user
                 query = (item.speakers.exclude(end_time=None)
```

Comment:
This block should be less indended. I does not have to run inside the speakers loop.